docs/tutorial/2-tilt-up.md

```diff
@@ -0,0 +1,143 @@
+>
+> This project uses `Dockerfile`s with Docker as the build engine and `kubectl`-friendly YAML files.
+> But that's only a small subset of Tilt functionality.
+>
+> Even if you're using other technologies (e.g. Helm, CRDs, `podman`), we recommend starting here to learn the Tilt fundamentals.
+>
+> Once you're comfortable with how Tilt works, we've got comprehensive guides on `Tiltfile` authoring that cover these topics and much more!
+
+## Run `tilt up` for the First Time
+Tilt brings consistency to your development not only due to ensuring a reproducible dev-environment-as-code, but launching any Tilt project is the same with the `tilt up` command, so you always have a familiar starting point.
+`tilt up` starts the Tilt control loop, which we'll explore in more detail in a moment.
+
+Navigate to the [`tilt-avatars` directory][tutorial-prerequisites-sample-project] in a terminal and launch Tilt:
+```bash
+tilt up --port=3366
+```
+
+> We've specified `--port=3366` here to free up the default port in case you want to launch another Tilt project simultaneously.
+> If not overridden, the default Tilt port is `10350`.
+
+You should see output similar to the following in your terminal:
+![Running tilt up in a Terminal window shows "Tilt started on http://localhost:3366/" message](/assets/docimg/tutorial/tilt-up-cli.gif)
+
+Hit `Spacebar` while your terminal is active, and Tilt will launch your default browser.
+(Or navigate to [http://localhost:3366/]() directly in your preferred browser.)
+
+In the next section, we'll explain the Tilt UI. But first, let's dissect what's happening in the background.
+<!-- TODO(milas): this would be a great place for a cheeky graphic about how we're stalling while the builds happen -->
+
+## `Tiltfile`
+When you run `tilt up`, Tilt looks for a special file named `Tiltfile` in the current directory, which defines your dev-environment-as-code.
+
+A `Tiltfile` is written in [Starlark][starlark], a simplified dialect of Python.
+
+> 🐍 Not a Python expert? No worries. Our guides have lots of examples, so you can copy & paste your way to success!
+
+Because your `Tiltfile` is a program, you can configure it with familiar constructs like loops, functions, and arrays.
+This makes the `Tiltfile` more extensible than a configuration file format like JSON or YAML, which requires hard-coding all possible options upfront.
+
+When Tilt executes the `Tiltfile`:
+ 1. Built-in functions like [`k8s_yaml`][api-k8s_yaml] and [`docker_build`][api-docker_build] register information with the Tilt engine
+ 2. Tilt uses the resulting configuration to assemble resources to build and deploy
+ 3. Tilt watches **relevant** source code files so it can trigger an update of the associated resource(s) 
+
+Within Tilt, the `Tiltfile` is itself a resource, so **you can even modify your `Tiltfile` and see the changes without restarting Tilt**!
+
+![Sample Tiltfile code](/assets/docimg/tutorial/tiltfile.png){:class="no-shadow"}
```

Comment:
Haaa so it looks like nothing else because this was a quick experiment at trying to provide some "interesting" visuals in the docs, which I don't feel was hugely successful due to my weak visual design skills if I'm honest 😓 

So, if we go further down this road, that's definitely worth doing...but we should probably also have a better looking design/process.